concurrent_test.go

```diff
@@ -321,3 +423,89 @@ func testResultsDirectory(t *testing.T) string {
 
 	return path
 }
+
+/*
+*********************************************************
+Data structures and functions for analyzing history records
```

Comment:
Introducing code section just means it should be a separate file/package.

Reply:
exactly. That's the plan. But in separate PR.